test/e2e-go/restAPI/restClient_test.go

```diff
@@ -2172,141 +2189,1102 @@ func TestMaxDepthAppWithPCTrace(t *testing.T) {
 
 	// Check expected == actual
 	creationOpcodeTrace := []model.SimulationOpcodeTraceUnit{
-		{Pc: 1},
-		{Pc: 6},
-		{Pc: 8},
-		{Pc: 9},
-		{Pc: 10},
-		{Pc: 149},
-		{Pc: 150},
-	}
-
-	recursiveLongOpcodeTrace := []model.SimulationOpcodeTraceUnit{
-		{Pc: 1},
-		{Pc: 6},
-		{Pc: 8},
-		{Pc: 9},
-		{Pc: 10},
-		{Pc: 13},
-		{Pc: 15},
-		{Pc: 16},
-		{Pc: 17},
-		{Pc: 21},
-		{Pc: 23},
-		{Pc: 25},
-		{Pc: 27},
-		{Pc: 29},
-		{Pc: 31},
-		{Pc: 33},
-		{Pc: 35},
-		{Pc: 37},
-		{Pc: 39},
-		{Pc: 41},
-		{Pc: 43},
-		{Pc: 45},
-		{Pc: 47},
-		{Pc: 48},
-		{Pc: 50},
-		{Pc: 51},
-		{Pc: 53},
-		{Pc: 54},
-		{Pc: 56},
-		{Pc: 59},
-		{Pc: 60},
-		{Pc: 61},
-		{Pc: 62},
-		{Pc: 63},
-		{Pc: 66},
-		{Pc: 67},
-		{Pc: 68},
-		{Pc: 69},
-		{Pc: 74},
-		{Pc: 75},
-		{Pc: 76},
-		{Pc: 78},
-		{Pc: 79},
-		{Pc: 81},
-		{Pc: 83},
-		{Pc: 85},
-		{Pc: 87},
-		{Pc: 89, SpawnedInners: &[]uint64{0}},
-		{Pc: 90},
-		{Pc: 91},
-		{Pc: 92},
-		{Pc: 94},
-		{Pc: 95},
-		{Pc: 97},
-		{Pc: 99},
-		{Pc: 103},
-		{Pc: 104},
-		{Pc: 106},
-		{Pc: 113},
-		{Pc: 116},
-		{Pc: 117},
-		{Pc: 118},
-		{Pc: 119},
-		{Pc: 121},
-		{Pc: 122},
-		{Pc: 123},
-		{Pc: 125},
-		{Pc: 128},
-		{Pc: 129},
-		{Pc: 130},
-		{Pc: 131},
-		{Pc: 132},
-		{Pc: 134},
-		{Pc: 136},
-		{Pc: 138},
-		{Pc: 139},
-		{Pc: 141},
-		{Pc: 143},
-		{Pc: 145, SpawnedInners: &[]uint64{1, 2}},
-		{Pc: 146},
-		{Pc: 72},
-		{Pc: 73},
-	}
-
-	finalDepthTrace := []model.SimulationOpcodeTraceUnit{
-		{Pc: 1},
-		{Pc: 6},
-		{Pc: 8},
-		{Pc: 9},
-		{Pc: 10},
-		{Pc: 13},
-		{Pc: 15},
-		{Pc: 16},
-		{Pc: 17},
-		{Pc: 21},
-		{Pc: 23},
-		{Pc: 25},
-		{Pc: 27},
-		{Pc: 29},
-		{Pc: 31},
-		{Pc: 33},
-		{Pc: 35},
-		{Pc: 37},
-		{Pc: 39},
-		{Pc: 41},
-		{Pc: 43},
-		{Pc: 45},
-		{Pc: 47},
-		{Pc: 48},
-		{Pc: 50},
-		{Pc: 51},
-		{Pc: 53},
-		{Pc: 54},
-		{Pc: 56},
-		{Pc: 59},
-		{Pc: 60},
-		{Pc: 61},
-		{Pc: 62},
-		{Pc: 63},
-		{Pc: 66},
-		{Pc: 67},
-		{Pc: 68},
-		{Pc: 69},
-		{Pc: 72},
-		{Pc: 73},
+		{
+			Pc: 1,
+		},
+		// txn ApplicationID
+		{
+			Pc: 6,
+			Additions: &[]model.StackValue{
```

Comment:
Feel free to ignore this, but it seems like it might be convenient to have helper functions to create StackValues, e.g.

```go
func StackValueInt(uint64) model.StackValue { ... }
func StackValueBytes([]byte) model.StackValue { ... }
```

Reply:
see [succinct restclient testcase](https://github.com/algorand/go-algorand/pull/5440/commits/0494b19f966ea3797fe3186d6e9542363a023771), I actually make the testcase more succinct with some `interface{}` trick